packages/core/contracts/oracle/implementation/VotingV2.sol

```diff
@@ -181,30 +182,31 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
         address indexed voter,
         address indexed caller,
         uint256 roundId,
+        uint256 priceRequestIndex,
         bytes32 indexed identifier,
         uint256 time,
-        int256 price,
         bytes ancillaryData,
+        int256 price,
         uint256 numTokens
     );
 
     event PriceRequestAdded(
         address indexed requester,
         uint256 indexed roundId,
+        uint256 priceRequestIndex,
```

Comment:
nit: why do we need to emit this variable in all these events? I thought of this more as an internal variable rather than something a frontend or end-user would be interested in?

Also, as a side note: does this get re-emitted when the vote rolls? If not, would this index be wrong in some cases?

Reply:
we use this in the front end in some places to track a request, this was the main reason I added it. Example screenshot below shows the "Dispute no"
![2022-09-13 at 11 43 40](https://user-images.githubusercontent.com/12886084/189869045-b6bbb34a-8dcf-4fc3-b564-fd867c120ba1.png)


Regarding rolled votes I think the right thing to do is always query the `skippedRequestIndexes` to see if there is a jump from the provided `priceRequestIndex` such that you _always_ show the _highest_ price request index in the front end (the most recent rolled request).

Reply:
Hmmm, does that work, though? Jumping would land you on an arbitrary request, right? Not necessarily the new location of that request, right?

For instance, if I had two rolled votes in a row, they would both show me the same number. I totally hear you that having some sort of unique identifier for a request to use in the frontend would be nice, but this number (at least from the price request added event) doesn't seem to be correct.

Would it be more correct to find the index by looking up the price request via the priceRequests mapping (by its request hash) and then grabbing the `.priceRequestIndex` from there, which should always be up to date?

Reply:
ye, you're right @mrice32 .

One counter I'll place is that in all non-rolled cases this number _is_ useful. if, however, a request is rolled then fetch the "correct" number via the `priceRequests` mapping. The front end already has logic for checking if a request is rolled. we can then use this index like this:
1) if not rolled then can directly use `priceRequestIndex` from the `PriceRequestAdded` Event.
2) if rolled then you need to do some extra state fetching and get the `.priceRequestIndex` from the `priceRequests` mapping for a given request. This requires an extra on-chain call per request which can be avoided in the first case, if not rolled which happens the vast majority of the time.

Reply:
How does it check the roll? Are we sure it doesn't need to do some sort of state call anyway?

Either way, this is fine, I just want to make sure we need the parameter.

Reply:
it checks if a vote was rolled by seeing if the current round is higher than the `lastVotingRound` & the vote was not settled by `pendingRequestIndex!=UINT64_MAX`.